server/eclipse-project/src/main/java/us/freeandfair/corla/json/ContestJsonAdapter.java

```diff
@@ -0,0 +1,178 @@
+   */
+  private static final String VOTES_ALLOWED = "votes_allowed";
+  
+  /**
+   * The "sequence number" string (for JSON serialization).
+   */
+  private static final String SEQUENCE_NUMBER = "sequence_number";
+  
+  /**
+   * Writes a contest object.
+   * 
+   * @param the_writer The JSON writer.
+   * @param the_info The object to write.
+   */ 
+  @Override
+  public void write(final JsonWriter the_writer, 
+                    final Contest the_contest) 
+      throws IOException {
+    the_writer.beginObject();
+    the_writer.name(ID).value(the_contest.id());
+    the_writer.name(NAME).value(the_contest.name());
+    the_writer.name(COUNTY_ID).value(the_contest.county().id());
+    the_writer.name(DESCRIPTION).value(the_contest.description());
+    the_writer.name(CHOICES);
+    the_writer.beginArray();
+    for (final Choice c : the_contest.choices()) {
+      the_writer.jsonValue(Main.GSON.toJson(Persistence.unproxy(c)));
+    }
+    the_writer.endArray();
+    the_writer.name(VOTES_ALLOWED).value(the_contest.votesAllowed());
+    the_writer.name(SEQUENCE_NUMBER).value(the_contest.sequenceNumber());
+    the_writer.endObject();
+  }
+  
+  /**
+   * Reads a contest object. 
+   * 
+   * @param the_reader The JSON reader.
+   * @return the object.
+   */
+  @Override
+  public Contest read(final JsonReader the_reader) 
+      throws IOException {
+    boolean error = false;
+    Long contest_id = null;
+    String contest_name = null;
+    County county = null;
+    String description = null;
+    Choice[] choices = null;
+    Integer votes_allowed = null;
+    Integer sequence_number = null;
+    
+    the_reader.beginObject();
+    while (the_reader.hasNext()) {
+      final String name = the_reader.nextName();
+      switch (name) {
+        case ID:
+          contest_id = the_reader.nextLong();
+          break;
+        
+        case NAME:
+          contest_name = the_reader.nextString();
+          break;
+          
+        case COUNTY_ID:
+          county = Persistence.getByID(the_reader.nextLong(), County.class);
+          break;
+          
+        case DESCRIPTION:
+          description = the_reader.nextString();
+          break;
+          
+        case CHOICES:
+          choices = Main.GSON.fromJson(the_reader.nextString(), Choice[].class);
+          break;
+          
+        case VOTES_ALLOWED:
+          votes_allowed = the_reader.nextInt();
+          break;
+          
+        case SEQUENCE_NUMBER:
+          sequence_number = the_reader.nextInt();
+          break;
+          
+        default:
+          error = true;
+          break;
+      }
+    }
+    the_reader.endObject();
+    
+    // check if an identically-numbered contest exists
+    
+    if (error || contest_id == null || county == null || description == null ||
+        choices == null || votes_allowed == null) {
+      throw new JsonSyntaxException("invalid data in contest");
```

Comment:
Would a `JsonParseException` be more appropriate? From the docs:

> This exception is raised if there is a serious issue that occurs during parsing of a Json string. One of the main usages for this class is for the Gson infrastructure. If the incoming Json is bad/malicious, an instance of this exception is raised.

Reply:
Perhaps. We do catch it in most places (it's a parent exception of `JsonSyntaxException`).